script-to-terminate-processes-by-port.md

```diff
@@ -0,0 +1,72 @@
+# Script to terminate a running task on a port
+## Prerequisites
+- `sudo` privileges for executing commands as root
+
+## Setup
+1. **Save the script**: Save the provided script to a file, for example, `terminate_process_on_port.sh`.
+
+    ```bash
+    #!/usr/bin/env bash
+
+    # Prompt for sudo privileges at the beginning of the script
+    if [ "$EUID" -ne 0 ]; then
+        sudo "$0" "$@"
```

Comment:
Redundant sudo Usage: Once the script is running with root privileges, additional sudo commands are unnecessary and could be removed I think

Reply:
Yep the whole script gets escalated, no need for `sudo` after that...